lmfit/minimizer.py

```diff
@@ -647,16 +645,17 @@ def penalty(self, fvars):
             sum-of-squares, but can be replaced by other options.
 
         """
-        if self.result.method in ['brute', 'shgo', 'dual_annealing']:
-            apply_bounds_transformation = False
-        else:
-            apply_bounds_transformation = True
+        apply_bounds_transformation = self.result.method not in [
+            'brute',
+            'shgo',
+            'dual_annealing',
+        ]
 
         r = self.__residual(fvars, apply_bounds_transformation)
         if isinstance(r, np.ndarray) and r.size > 1:
             r = self.reduce_fcn(r)
-            if isinstance(r, np.ndarray) and r.size > 1:
-                r = r.sum()
+        if isinstance(r, np.ndarray) and r.size > 1:
+            r = r.sum()
```

Comment:
I don't see a benefit here and am sure the changed code will work as well; to me the flow is visually easier to follow in the old code; the second check/manipulation will only happen if the first criteria is met. I'd prefer to revert.